packages/editor/src/components/local-autosave-monitor/index.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * External dependencies
+ */
+import { once, uniqueId, omit } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect, useRef } from '@wordpress/element';
+import { ifCondition } from '@wordpress/compose';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { parse } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import AutosaveMonitor from '../autosave-monitor';
+
+/**
+ * Time interval at which local autosave should occur after a change, in
+ * seconds.
+ *
+ * @var {number}
+ */
+const AUTOSAVE_INTERVAL_SECONDS = 15;
+
+/**
+ * Function which returns true if the current environment supports browser
+ * localStorage, or false otherwise. The result of this function is cached and
+ * reused in subsequent invocations.
+ */
+const hasLocalStorageSupport = once( () => {
+	try {
+		// Private Browsing in Safari 10 and earlier will throw an error when
+		// attempting to set into localStorage. The test here is intentional in
+		// causing a thrown error as condition bailing from local autosave.
+		window.localStorage.setItem( '__wpEditorTestLocalStorage', '' );
+		window.localStorage.removeItem( '__wpEditorTestLocalStorage' );
+		return true;
+	} catch ( error ) {
+		return false;
+	}
+} );
+
+/**
+ * Custom hook which returns a callback function to be invoked when a local
+ * autosave should occur.
+ *
+ * @return {Function} Callback function.
+ */
+function useAutosaveSaveCallback() {
```

Comment:
`useAutosaveCallback`?

Reply:
> Should this be `useAutosaveCallback`?

Yeah, it might read a little better. I think I chose it on the basis of being a specific sort of save callback.

>If we are explicitly including the local storage action, maybe `useAutosaveStore` could work too.

Sure. We might want to consider if we want any naming conventions around values assigned from hooks, e.g. in this case `useCallback`.

Reply:
> Sure. We might want to consider if we want any naming conventions around values assigned from hooks, e.g. in this case `useCallback`.

That's good enough for me, at least until we have proper conventions. Let's stick with `useAutosaveCallback`.